modules/eks/submodules/irsa/variables.tf

```diff
@@ -0,0 +1,128 @@
+
+variable "eks_info" {
```

Comment:
seems like a lot of detail.
one could pass the actual resources through or just query them as data, no?

Reply:
we can not query the data if we want to provision in the global/commercial account. In addtion we treat the var as an object and just pass it around, rather than tailoring each variable.

Reply:
do we actually use all these values?
this should only require exactly what is needed, not what can easily be passed down.

Reply:
We are using the same approach everywhere, where the module outputs are treated like objects and passed around, I am not convinced there is an actual drawback to that, if it is then we will need to discuss it on its merits not as part of this.

Reply:
After thinking about it more, what you said makes sense, given this could be used outside the module there is no reason to be buried, other than a lose relationship with eks.